src/test/java/LoginActionTest.java

```diff
@@ -0,0 +1,41 @@
+import org.junit.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class LoginActionTest {
+    @Test
+    public void testCheckUsernameSuccess(){
+        LoginAction userInput = new LoginAction("Herman1", "Password");
+        Assertions.assertTrue(userInput.checkUsername());
+    }
+
+    @Test
+    public void testCheckUsernameSuccessDifferentPassword(){
+        LoginAction userInput = new LoginAction("Herman1", "Password1");
+        Assertions.assertTrue(userInput.checkUsername());
+    }
+
+    @Test
+    public void testCheckUsernameFail(){
+        LoginAction userInput = new LoginAction("Herman2", "Password");
+        Assertions.assertFalse(userInput.checkUsername());
+    }
+
+    @Test
+    public void testCheckUserMatchesPasswordSuccess(){
+        LoginAction userInput = new LoginAction("Herman1", "Password");
+        Assertions.assertTrue(userInput.checkUserMatchesPassword());
+    }
+
+    @Test
+    public void testCheckUserMatchesPasswordWrongPassword(){
+        LoginAction userInput = new LoginAction("Herman1", "password");
+        Assertions.assertFalse(userInput.checkUserMatchesPassword());
+    }
+
+    @Test
+    public void testCheckUserMatchesPasswordUserDNE(){
+        LoginAction userInput = new LoginAction("Herman", "password");
```

Comment:
The test for checking if the user doesn't exist works for now, however we could run into an issue where someone creates a user Herman and then this test would fail. I suggest using a name that we wouldn't create a user in such as "TestingUserDNEName"? Or something like that. Otherwise good job on the multiple test cases!

Reply:
Sounds good, thanks Herman!